File: src/main/java/leetcode/medium/add_two_numbers/ListNode.java
package leetcode.medium.add_two_numbers;

public class ListNode {
    int val;
    ListNode next;

    ListNode() {
    }

    ListNode(int val) {
        this.val = val;
    }

    ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }

    @Override
    public String toString() {
        String res = String.valueOf(val);
        ListNode refList = this.next;
        while (refList != null) {
            res = res.concat(String.valueOf(refList.val));
            refList = refList.next;
        }
        return res;
    }
}

File: src/main/java/leetcode/hard/list_node/ListNode.kt
package leetcode.hard.list_node

class ListNode(var `val`: Int) {
    var next: ListNode? = null
    override fun toString(): String {
        var res = `val`.toString()
        var refList: ListNode? = next
        while (refList != null) {
            res += refList.`val`.toString()
            refList = refList.next
        }
        return res
    }
}